src/Components/WebAssembly/Authentication.Msal/src/MsalWebAssemblyServiceCollectionExtensions.cs

```diff
@@ -53,7 +53,7 @@ public static class MsalWebAssemblyServiceCollectionExtensions
             where TAccount : RemoteUserAccount
         {
             services.AddRemoteAuthentication<TRemoteAuthenticationState, TAccount, MsalProviderOptions>(configure);
-            services.TryAddEnumerable(ServiceDescriptor.Singleton<IPostConfigureOptions<RemoteAuthenticationOptions<MsalProviderOptions>>, MsalDefaultOptionsConfiguration>());
+            services.TryAddEnumerable(ServiceDescriptor.Scoped<IPostConfigureOptions<RemoteAuthenticationOptions<MsalProviderOptions>>, MsalDefaultOptionsConfiguration>());
```

Comment:
Is there state stored in MsalDefaultOptionsConfiguration?  Switching to IOptionsSnapshot will create a new options instance and rerun all of the Configure's each request

Reply:
Not each request. At least not in webassembly. In webassembly there is only the root scope. It changes when this package is used in something like Mobile Blazor Bindings, where there is a scope per webview.
The only reason it could be per request, is in prerendering on the server. But then you should want to have it created per request, as other types in the dependency tree are registered as scoped as well. Unfortunately that means that the options will be reread and recreated every request indeed.

Reply:
@HaoK I think this is fine, we want all these services to be registered as Scoped so that they can play well in the context of prerendering

Reply:
Ok fair enough, one final question/consideration before signing off.  @jspuij did you take a look at IOptionsMonitor and consider if that is applicable at all?  Per request options computation can get very expensive performance wise depending on what the configures are doing.  I'm not sure if there's any way to only recompute on demand here, but just something I wanted to make sure was considered

Reply:
The issue isn't option reload (I'm not after that), the issue is that DefaultOidcOptionsConfiguration is registered as singleton, while it takes a NavigationManager as a dependency, which is scoped in MBB and probably during prerendering of webassembly as well. Unfortunately IOptions<> will always resolve from the root scope, regardless of the lifetimes of any dependency that is injected in the tree.  IOptionSnapshot will not, but will of course load the options everytime it is resolved from the scope. 

Unfortunately by taking on a dependency on NavigationManager in [DefaultOidcProviderOptionsConfiguration](https://github.com/dotnet/aspnetcore/blob/master/src/Components/WebAssembly/WebAssembly.Authentication/src/Options/DefaultOidcProviderOptionsConfiguration.cs#L13), something that could be loaded and configured once, is now tied to the lifetime of the dependency. (It only needs a base address, not an entire navigation manager). This could probably be redesigned, but not without breaking the public API.

Reply:
Okay, makes sense, thanks for the clarification